src/test/java/me/karakelley/http/server/HttpRequestReaderTest.java

```diff
@@ -85,9 +91,37 @@ void testWhenABodyIsPresent() {
     assertEquals("hey", new String(request.getBody()));
   }
 
+  @Test
+  void testThatIOExceptionIsCaughtAndLogged() {
+    withAppender(logger -> {
+      ByteArrayInputStream inputStream = new ByteArrayInputStream("GET / HTTP\r\n\r\n".getBytes());
+      HttpRequestReader requestReader = new HttpRequestReader(inputStream);
+      try {
+        inputStream.close();
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
If you're in a junit test and you're capturing an exception you don't expect to be thrown it's probably a better idea to do something like this...

```
try {
  // do the thing...
} catch (SomeException e) {
  fail(e);
}
```

You'll still get the error message printed via Junit, but your test will fail (which is better than silently passing for the wrong reason)

Reply:
Oh! Cool! I had seen that before but never understood why someone would want their test to fail - seeing it in this context makes much more sense.